### PR TITLE
Refresh the gem and publish to rubygems.org

### PR DESCRIPTION
**Pull request by jfeltesse-mdsol**
Taking over https://github.com/mdsol/aranea/pull/9 which has been sitting for a few weeks in draft and is incomplete anyway.

## Why?
Without automatically pushing to rubygems.org, this gem is unnecessarily forcing people to pull from github. Also, it starts to show its age and will blow up in the face of anyone who uses Faraday 1.x for instance.
I added the rubygems.org token to the travis settings page and manually published the previous version, `0.0.3`.

## Included
- require ruby 2.5+
- require active_support 4.2+
- publish to rubygems.org upon tagging
- support faraday 0.x and 1.x
- run travis with ruby 2.5, 2.6, 2.7, 3.0
- fix style violations using https://github.com/mdsol/rubocop-mdsol
- removed Timecop in favor of ActiveSupport built-in time helpers.

## Not included
- proper Appraisal combinations to test with different activesupport/faraday/ruby versions
  - an exercise left to the reader who actually uses this gem
- proper adhoc-ing.
  - I don't use the gem myself

@johnzhangmdsol

### CODE REVIEW

File: lib/aranea/failure_repository.rb [hunk 1]
```diff
@@ -58,17 +53,15 @@ def respond!
     end
 
     def to_s
-      "Failure on #{@pattern.inspect} ending at approximately #@expiration_date"
+      "Failure on #{@pattern.inspect} ending at approximately #{@expiration_date}"
```
Comment (ykitamura-mdsol) [inline lib/aranea/failure_repository.rb:56]: 😲

**Review by ykitamura-mdsol — COMMENTED** [review]
looks good to me 👍

**Review by johnzhangmdsol — APPROVED** [review]
I had no idea others were using v0.0.3! Thank you @jfeltesse-mdsol - merging by 12 ET

File: lib/aranea/failure_repository.rb (hunk 1 above)
Comment (jfeltesse-mdsol) [reply lib/aranea/failure_repository.rb:56]: ikr 🙈

**Comment by jfeltesse-mdsol** [comment]
> I had no idea others were using v0.0.3! 

Am not, I'm doing this because it's weird in 2021 this gem still needs to be pushed manually to rubygems.org and has no CI. I had already done similar work on https://github.com/mdsol/rack_content_type_default/pull/7 so it's not like it took me very long to put that PR together.

**Pull request merged**